hamocc/mo_apply_rivin.F90

```diff
@@ -97,63 +99,113 @@ subroutine apply_rivin(kpie,kpje,kpke,pddpo,omask,rivin)
-               &                          + rivin(i,j,irdip)*fdt/volij
-          ocetra(i,j,1:kmle(i,j),ialkali) = ocetra(i,j,1:kmle(i,j),ialkali)                        &
-               &                          + rivin(i,j,iralk)*fdt/volij
-          if (use_natDIC) then
-            ocetra(i,j,1:kmle(i,j),inatsco212) = ocetra(i,j,1:kmle(i,j),inatsco212)                &
-                 &                             + rivin(i,j,iralk)*fdt/volij                        &
-                 &                             + rivin(i,j,irdin)*fdt/volij                        &
-                 &                             + rivin(i,j,irdip)*fdt/volij
-            ocetra(i,j,1:kmle(i,j),inatalkali) = ocetra(i,j,1:kmle(i,j),inatalkali)                &
-                 &                             + rivin(i,j,iralk)*fdt/volij
+          if (use_r2o) then
+            ! Riverine labile DOC (riv_lDOC) instantaneously degraded as DIC.
+            ! DIC <= riv_DIC + riv_lDOC
+            ! Riverine DIN and DIP from remineralized riv_lDOC are already included in the
+            ! dataset using C:N:P 2583:103:1.
+            ! Riverine semi-labile DOC (riv_slDOC, high C content) goes in high-carbon terrestrial
+            ! DOC (tDOC_hc).
+            ! tDOC_hc <= riv_slDOC
+            ! Riverine POC instantaneously dissolved as low-carbon tDOC (tDOC_lc)
+            ! tDOC_lc <= riv_POC
+            ocetra(i,j,1:kmle(i,j),iano3)   = ocetra(i,j,1:kmle(i,j),iano3)                          &
+                 &                          + rivin(i,j,irdin)*fdt/volij
+            ocetra(i,j,1:kmle(i,j),iphosph) = ocetra(i,j,1:kmle(i,j),iphosph)                        &
+                 &                          + rivin(i,j,irdip)*fdt/volij
+            ocetra(i,j,1:kmle(i,j),isilica) = ocetra(i,j,1:kmle(i,j),isilica)                        &
+                 &                          + rivin(i,j,irsi) *fdt/volij
+            ocetra(i,j,1:kmle(i,j),iiron)   = ocetra(i,j,1:kmle(i,j),iiron)                          &
+                 &                          + rivin(i,j,iriron)*fdt/volij*dFe_frac
+            ocetra(i,j,1:kmle(i,j),itdoc_hc) = ocetra(i,j,1:kmle(i,j),itdoc_hc)                      &
+                 &                          + rivin(i,j,irtdoc)*fdt/volij
+            ocetra(i,j,1:kmle(i,j),itdoc_lc) = ocetra(i,j,1:kmle(i,j),itdoc_lc)                      &
+                 &                          + rivin(i,j,irdet)*fdt/volij
+            ocetra(i,j,1:kmle(i,j),isco212) = ocetra(i,j,1:kmle(i,j),isco212)                        &
+                 &                          + rivin(i,j,iralk)*fdt/volij                             &
+                 &                          + rivin(i,j,irdoc)*fdt/volij
+            ocetra(i,j,1:kmle(i,j),ialkali) = ocetra(i,j,1:kmle(i,j),ialkali)                        &
+                 &                          + rivin(i,j,iralk)*fdt/volij
```

Comment:
I vaguely remember that the alkalinity shouldn't be changed due to immediate remineralization of detritus during riverine input according to the protocol, right? - I feel it is worth mentioning this as a comment (also affecting the alkalinity budget, causing an imbalance). Further, I would suggest to make the separation between the old and new scheme more obvious by only switching for the tracers, that are affected (i.e. NO3, PO4, silica and Fe + alkali? seem to be the same - and could be kept out of the if-else statement).